pkg/agent/baker.go

```diff
@@ -379,7 +379,10 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return IsKubeletConfigFileEnabled(cs, profile, config.EnableKubeletConfigFile)
 		},
 		"IsKubeletClientTLSBootstrappingEnabled": func() bool {
-			return IsKubeletClientTLSBootstrappingEnabled(config.KubeletClientTLSBootstrapToken)
+			return config.EnableSecureTLSBootstrapping || IsKubeletClientTLSBootstrappingEnabled(config.KubeletClientTLSBootstrapToken)
```

Comment:
since we're OR'ing, EnableSecureTLSBootstrapping being false won't affect the overall value

Reply:
clueless question - whats the difference between config having EnableSecureTLSBootstrapping set to true and IsKubeletClientTLSBootstrappingEnabled returning true?

Reply:
basically, `EnableSecureTLSBootstrap` will be true when we are doing TLS bootstrapping but we're NOT using a hard-coded token, `IsKubeletClientTLSBootstrappingEnabled` will be true when the node bootstrapping config we get from RP includes a hard-coded token, also implying that we're doing TLS bootstrapping.

so `IsKubeletClientTLSBootstrappingEnabled` overall will return true when we're doing _some_ form of TLS bootstrapping, whether that be the secure way or with a hard-coded token

Reply:
Thanks, makes sense - might be a good idea to add this as a comment somewhere maybe for the future reference?

Reply:
did some renaming to make this a bit more clear, also added comments :)